appender.go

```diff
@@ -9,24 +9,85 @@ import (
 	"database/sql/driver"
 	"errors"
 	"fmt"
+	"reflect"
 	"time"
 	"unsafe"
 )
 
-// Appender holds the DuckDB appender. It allows to load bulk data into a DuckDB database.
+const errNote = "NOTE: this is based on types initialized from the first row of appended data. Please confirm this matches the schema."
+
+var typeIdMap = map[C.duckdb_type]string{
+	C.DUCKDB_TYPE_BOOLEAN:   "bool",
+	C.DUCKDB_TYPE_TINYINT:   "int8",
+	C.DUCKDB_TYPE_SMALLINT:  "int16",
+	C.DUCKDB_TYPE_INTEGER:   "int32",
+	C.DUCKDB_TYPE_BIGINT:    "int64",
+	C.DUCKDB_TYPE_UTINYINT:  "uint8",
+	C.DUCKDB_TYPE_USMALLINT: "uint16",
+	C.DUCKDB_TYPE_UINTEGER:  "uint32",
+	C.DUCKDB_TYPE_UBIGINT:   "uint64",
+	C.DUCKDB_TYPE_FLOAT:     "float32",
+	C.DUCKDB_TYPE_DOUBLE:    "float64",
+	C.DUCKDB_TYPE_VARCHAR:   "string",
+	C.DUCKDB_TYPE_BLOB:      "[]uint8",
+	C.DUCKDB_TYPE_TIMESTAMP: "time.Time",
+	C.DUCKDB_TYPE_UUID:      "duckdb.UUID",
+	C.DUCKDB_TYPE_LIST:      "slice",
+	C.DUCKDB_TYPE_STRUCT:    "struct",
+}
+
+// SetColValue is the type definition for all column callback functions.
+type SetColValue func(a *Appender, info *colInfo, rowIdx C.idx_t, val any)
+
+// colInfo holds a column's underlying vector, a callback function to write column values to this vector, and additional helper fields.
+type colInfo struct {
+	vector C.duckdb_vector
+	fn     SetColValue
+
+	// The type of the column.
+	ddbType C.duckdb_type
```

Comment:
I would call it `duckdbType` or `dbType` as it is more clear than `ddbType`. Reads like a typo.